packages/arb-token-bridge-ui/src/token-bridge-sdk/utils.ts

```diff
@@ -100,26 +100,63 @@ export const getBridger = async ({
     : Erc20Bridger.fromProvider(destinationChainProvider)
 }
 
-const getProviderForChainCache: {
+const providerInstanceCache: {
   [chainId: number]: StaticJsonRpcProvider
 } = {
   // start with empty cache
 }
 
-function createProviderWithCache(chainId: ChainId) {
+const enableCaching = (chainId: number) => {
+  const txReceiptsCachingEnabledConfig =
+    process.env.NEXT_PUBLIC_PROVIDER_CACHE_TX_RECEIPTS || 'testnet,mainnet' // default to 'testnet,mainnet' if not set
+
+  return txReceiptsCachingEnabledConfig.includes(
+    isNetwork(chainId).isTestnet ? 'testnet' : 'mainnet'
+  )
+}
+
+const enableBatching = (chainId: number) => {
+  const rpcRequestBatchingEnabledConfig =
+    process.env.NEXT_PUBLIC_PROVIDER_BATCH_RPC || 'testnet' // default to 'testnet' if not set
```

Comment:
why don't we want to enable `mainnet` batching by default?

Reply:
We will enable `mainnet` after we confirm stability for testnets in the production build first. We did the same for caching.